copilot-theorem/src/Copilot/Theorem/What4/Translate.hs

```diff
@@ -475,14 +504,15 @@ translateExpr sym localEnv e offset = case e of
 
 type BVOp1 sym w = (KnownNat w, 1 <= w) => WI.SymBV sym w -> IO (WI.SymBV sym w)
 
-type FPOp1 sym fpp =
-  KnownRepr WT.FloatPrecisionRepr fpp =>
-  WI.SymExpr sym (WT.BaseFloatType fpp) -> IO (WI.SymExpr sym (WT.BaseFloatType fpp))
+type FPOp1 sym fi =
+  WFP.FloatInfoRepr fi ->
```

Comment:
Note that I've changed the various `FPOp*` types to have an explicit `FloatInfoRepr` argument. This was done to avoid the need for `AllowAmbiguousTypes`, as `fi` would be ambiguous otherwise. On the other hand, several methods in the `IsInterpretedFloatExprBuilder` class are already ambiguous and require explicit type applications (e.g., `iFloatIte`), so if you'd prefer, I could just as well remove these `FloatInfoRepr` arguments.